src/frontend/types/libs/JitsiMeetExternalAPI/index.d.ts

```diff
@@ -0,0 +1,44 @@
+export as namespace JitsiMeetExternalAPI;
+export = JitsiMeetExternalAPI;
+
+type jistiCommand = 'startRecording' | 'stopRecording';
+type jitsiRecordingMode = 'stream' | 'file';
+type jitsiRecordingOptions = {
+  mode: jitsiRecordingMode;
+  rtmpStreamKey: string;
+};
+
+declare class JitsiMeetExternalAPI {
+  constructor(
+    domain: string,
+    options?: {
+      configOverwrite?: {
+        constraints?: {
+          video: {
+            height: {
+              ideal: number;
+              max: number;
+              min: number;
+            };
+          };
+        };
+        resolution?: number;
+        toolbarButtons?: string[];
+      };
+      interfaceConfigOverwrite?: {
+        HIDE_INVITE_MORE_HEADER?: boolean;
+        TOOLBAR_BUTTONS?: string[];
+      };
+      parentNode?: HTMLElement;
+      roomName?: string;
+    },
+  );
```

Comment:
This case remind me that I always wonder if I should at the creation of the type, write all attributes that compose it or just put them that I will used right now.

With the minimal approach, reader sees directly what attribute is used by our app but for a new comers having all the attribute can help him to know more quickly what is possible to do with this interface.

Reply:
Most of the time I create only what I need because I don't want to maintain it. IMO if we create the all defition we should do it in [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) repo don't you think ?

Reply:
Yes, I totaly agree!